subsys/settings/src/settings_ram.c

```diff
@@ -0,0 +1,293 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <stdbool.h>
+#include <zephyr.h>
+
+#include "settings_priv.h"
+
+#include <logging/log.h>
+#include <sys/mutex.h>
+#include <sys/sys_heap.h>
+#include <stdalign.h>
+
+LOG_MODULE_DECLARE(settings, CONFIG_SETTINGS_LOG_LEVEL);
+
+/* Alignment of memory blocks returned by malloc. */
+const static size_t kMallocAlignment = alignof(long long);
+uint8_t sHeapMemory[CONFIG_SETTINGS_RAM_HEAP_SIZE];
+struct sys_heap sHeap;
+SYS_MUTEX_DEFINE(heap_lock);
```

Comment:
Use the zephyr heap abstractions, no need for a mutex then.

```suggestion
static K_HEAP_DEFINE(ram_settings_heap, CONFIG_SETTINGS_RAM_HEAP_SIZE);
```